Backend/BlockChangesService.cs

```diff
@@ -93,7 +93,8 @@ private async Task RemoveStudentRequests(BlockChangeRequest request)
                                                      x.BlockFrom.CourseId == request.BlockFrom.CourseId &&
                                                      x.BlockFrom.StartHour == request.BlockFrom.StartHour &&
                                                      x.BlockFrom.Day == request.BlockFrom.Day &&
-                                                     x.BlockFrom.Duration == request.BlockFrom.Duration);
+                                                     x.BlockFrom.Duration == request.BlockFrom.Duration &&
+                                                     x.Status == ExchangeStatus.WaitingForExchange);
```

Comment:
Just small problem, the name of the method is not good... RemoveNorRealizedRequests or similar...